mslice/scripting/__init__.py

```diff
@@ -45,6 +45,17 @@ def generate_script_lines(raw_ws, workspace_name):
     ws_name = workspace_name.replace(".", "_")
     alg_history = raw_ws.getHistory().getAlgorithmHistories()
     existing_ws_refs = []
+
+    # Loop back from end to see if we have a Save / Load pair and truncate at the load if so
+    prev_algo = alg_history[-1]
+    for idx, algorithm in reversed(list(enumerate(alg_history[:-1]))):
```

Comment:
This is for when we load an `nxs` file with a long history from the data reduction which ends with a call to `SaveNexus` - If we put all this history into the script file it will not run, so this is to truncate it at a point in the history where a `Save*` is followed immediately by a `Load`. 

There was such a check in the code originally (which Andre put in) but I think I took it out by mistake when I was refactoring things, because I was mostly testing it on `nxspe` file where the history was lost.

Reply:
Ah yes I remember discussing that with Andre. Thanks for the explanation.